src/Http/Routing/src/Builder/RoutingEndpointConventionBuilderExtensions.cs

```diff
@@ -130,7 +130,7 @@ public static class RoutingEndpointConventionBuilderExtensions
         /// <returns>The <see cref="IEndpointConventionBuilder"/>.</returns>
         public static TBuilder WithName<TBuilder>(this TBuilder builder, string endpointName) where TBuilder : IEndpointConventionBuilder
         {
-            builder.WithMetadata(new EndpointNameAttribute(endpointName));
+            builder.WithMetadata(new EndpointNameAttribute(endpointName), new RouteNameMetadata(endpointName));
```

Comment:
I see that both [IEndpointNameMetadata](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.routing.iendpointnamemetadata?view=aspnetcore-3.0) and [IRouteNameMetadata](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.routing.iroutenamemetadata?view=aspnetcore-3.0) have been around in the same namespace and assembly since at least 3.0. The doc comments for both indicate that they're used for link-generation.

@pranavkm @JamesNK Do either of you know why we have a route name and an enpoint name and how they're used differently?

Reply:
Sorry, I don't know. I didn't touch link generation much. When I worked on routing I focused on endpoint generation and route matching.